src/credentials/activation.ts

```diff
@@ -31,15 +32,21 @@ export async function loginWithMostRecentCredentials(
     toolkitSettings: SettingsConfiguration,
     loginManager: LoginManager
 ): Promise<void> {
+    const profiles = await CredentialsProviderManager.getInstance().getProfiles()
+    const profileNames = Object.keys(profiles)
     const previousCredentialsId = toolkitSettings.readSetting<string>(profileSettingKey, '')
-    if (previousCredentialsId) {
+    if (profiles && profileNames.length === 1) {
+        // Auto-connect if there is exactly one profile, named "default".
+        await loginManager.login(profiles[profileNames[0]])
+        // Toast.
+        vscode.window.showInformationMessage(`Connected to "${profileNames[0]}"`)
```

Comment:
The message needs more context, as we're not necessarily the only extension a user has installed.
Perhaps "AWS Toolkit is now connected to AWS with {x}"